src/services/tx/tx-sender/sdk.ts

```diff
@@ -36,24 +37,29 @@ async function switchOrAddChain(walletProvider: ConnectedWallet['provider'], cha
       params: [{ chainId: hexChainId }],
     })
   } catch (error) {
-    if ((error as Error & { code: number }).code !== UNKNOWN_CHAIN_ERROR_CODE) {
-      throw error
+    const errorCode = get(error, 'code') as number | undefined
+
+    // Rabby emits the same error code as MM, but it is nested
+    const nestedErrorCode = get(error, 'data.originalError.code') as number | undefined
```

Comment:
Is this better than `data?.originalError?.code`?

Reply:
I think it is definitely more readable. As an alternative I tried to check for each key before accessing it i.e.`'data' in error && 'originalError' in (error.data as ...) && error.data.originalError` but it got messy really fast. We could ofc also go with `data?.originalError?.code` and ts-ignore it.